app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneServerFactoryImpl.kt

```diff
@@ -8,14 +8,30 @@ package ch.epfl.sdp.drone3d.service.impl.drone
 import ch.epfl.sdp.drone3d.service.api.drone.DroneServerFactory
 import io.mavsdk.System
 import io.mavsdk.mavsdkserver.MavsdkServer
+import kotlinx.coroutines.*
+import org.jetbrains.annotations.Async
+import kotlin.concurrent.thread
 
 class DroneServerFactoryImpl: DroneServerFactory {
 
-    override fun createSimulation(simIP: String, simPort: String): DroneServerFactory.InstanceContainer {
+    override fun createSimulation(simIP: String, simPort: String): DroneServerFactory.InstanceContainer? {
         val mavsdkServer = MavsdkServer()
         val cloudSimIPAndPort = "tcp://${simIP}:${simPort}"
-        val mavsdkServerPort = mavsdkServer.run(cloudSimIPAndPort)
-        return DroneServerFactory.InstanceContainer(mavsdkServer, System("localhost", mavsdkServerPort))
+        var mavsdkServerPort = 0
+
+        //Tries to setup the server for 3 seconds
+        GlobalScope.launch {
+            launch {
+                mavsdkServerPort = mavsdkServer.run(cloudSimIPAndPort)
+            }
+        }
+        Thread.sleep(3000)
```

Comment:
I don't really like that, is there another way to fix it ?

Reply:
I could not find another way because to use corrupting with a timeout and delay, we would need mavsdkServer.run(cloudSimIPAndPort) to be a suspend function which it is not.